swift/ql/lib/codeql/swift/StringFormat.qll

```diff
@@ -72,3 +85,16 @@ class NsExceptionRaise extends FormattingFunction, Method {
 
   override int getFormatParameterIndex() { result = 1 }
 }
+
+/**
+ * A function that appears to be an imported C `printf` variant.
+ */
+class PrintfFormat extends FormattingFunction, FreeFunction {
```

Comment:
Should we check types as well as names?

Reply:
We could, though I didn't find we need to.

I suppose the real question is, if we have something matching `%printf%(...:format:...)` and the types _don't_ match (say because it's a custom `printf` implementation rather than an imported C header and the type is a `String` rather than the opaque or char buffer pointer or whatever exactly we expect), is this more likely to be a formatting function of some sort or something else?

I would say it's more likely to be a formatting function, so we want to match it anyway.

Reply:
FWIW, I agree with you @geoffw0. But I'd like to hear if @rdmarsh2 has any insights on this that we're missing.

Reply:
Yeah, there's a slight change of direction here that ultimately comes from the initiative to address false negatives.  Previously we would always have sought to be as precise as possible - whereas now we are OK with a heuristic / imprecise approach that might catch more results.  The quality of matches still has to be high, which in practice (for all of the PRs I've been working on this iteration) has meant reviewing long lists of sinks on MRVA, tuning the rules, and reviewing final query results as well.  Happy to discuss any aspects of this in more detail.